### PR TITLE
Fix core.wrap_text

### LINKED ISSUE #6234: minetest.wrap_text is broken
`minetest.wrap_test` looks very broken to me, it returns very weird result in MT 0.4.16.

I used `luacmd` to test.

This function returns a *table* of strings. As if the source string has been splitted. There are no newlines. And the strings can even be *longer* than the length limit. This output is not useful at all.
Proof:
My mama tell me
https://i.imgur.com/J40l346.png

To test for yourself, just try to call `minetest.wrap_text` with help of the `luacmd` and experiment a bit. You will see what happens.

### PR DESCRIPTION
**Pull request by sfan5**
fixes #6234 because apparently nobody can be bothered

### CODE REVIEW

**Comment by raymoo** [comment]
Will this work even if a word is wider than the requested text width?

**Comment by sfan5** [comment]
Yes.

File: builtin/common/misc_helpers.lua [hunk 1]
```diff
@@ -308,59 +308,25 @@ function core.formspec_escape(text)
...
+	local line = {}
+	if #text <= max_length then
+		return as_table and {text} or text
+	end
+
+	for word in text:gmatch('%S+') do
+		local cur_length = #table.concat(line, ' ')
+		if cur_length > 0 and cur_length + #word + 1 >= max_length then
+			-- word wouldn't fit on current line, move to next line
+			table.insert(result, table.concat(line, ' '))
```
Comment (SmallJoker) [inline builtin/common/misc_helpers.lua]: Words like
this_monstrous_extremely_long_example_word_to_demonstrate
that wouldn't fit on one line,
should be splitted to guarantee
that the limit isn't violated.

**Comment by SmallJoker** [comment]
Thanks for picking up this PR, I'm glad to see this solution for the mainmenu problem.
There's one problem I've found with longer words - apart from that it looks very simple and clean.

File: builtin/common/misc_helpers.lua (hunk 1 above)
Comment (sfan5) [reply builtin/common/misc_helpers.lua]: I don't think it's unreasonable *not* to cut words in half.
The intention of this function is not to absolutely enforce the limit in all cases (that's how i see it at least).

maybe noting this in the docs would suffice

**Comment by sfan5** [comment]
6fa2f6b4aa42683fc85a6ce4a2a03a4b47c36974

**Pull request closed without merging**